ballerina-tests/tests/request_interceptors_basic_test.bal

```diff
@@ -0,0 +1,240 @@
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+
+final http:Client defaultRequestInterceptorClientEP = check new("http://localhost:" + defaultRequestInterceptorTestPort.toString());
+
+listener http:Listener defaultRequestInterceptorServerEP = new(defaultRequestInterceptorTestPort, config = {
+        interceptors : [new DefaultRequestInterceptor(), new LastRequestInterceptor()]
+    });
+
+service / on defaultRequestInterceptorServerEP {
+
+    resource function 'default .(http:Caller caller, http:Request req) returns error? {
+        http:Response res = new();
+        res.setHeader("last-interceptor", check req.getHeader("last-interceptor"));
+        res.setHeader("default-interceptor", check req.getHeader("default-interceptor"));
+        res.setHeader("last-request-interceptor", check req.getHeader("last-request-interceptor"));
+        check caller->respond(res);
+    }
+}
+
+@test:Config{}
+function testDefaultRequestInterceptor() returns error? {
+    http:Response res = check defaultRequestInterceptorClientEP->get("/");
+    assertHeaderValue(check res.getHeader("last-interceptor"), "default-interceptor");
+    assertHeaderValue(check res.getHeader("default-interceptor"), "true");
+    assertHeaderValue(check res.getHeader("last-request-interceptor"), "true");
+
+    res = check defaultRequestInterceptorClientEP->post("/", "testMessage");
+    assertHeaderValue(check res.getHeader("last-interceptor"), "default-interceptor");
+    assertHeaderValue(check res.getHeader("default-interceptor"), "true");
+    assertHeaderValue(check res.getHeader("last-request-interceptor"), "true");
+}
+
+final http:Client requestInterceptorReturnsErrorClientEP = check new("http://localhost:" + requestInterceptorReturnsErrorTestPort.toString());
+
+listener http:Listener requestInterceptorReturnsErrorServerEP = new(requestInterceptorReturnsErrorTestPort, config = {
+        interceptors : [new DefaultRequestInterceptor(), new RequestInterceptorReturnsError(), new LastRequestInterceptor()]
+    });
+
+service / on requestInterceptorReturnsErrorServerEP {
+
+    resource function 'default .() returns string {
+        return "Response from resource - test";
+    }
+}
+
+@test:Config{}
+function testrequestInterceptorReturnsError() returns error? {
+    http:Response res = check requestInterceptorReturnsErrorClientEP->get("/");
+    test:assertEquals(res.statusCode, 500);
+    assertTextPayload(check res.getTextPayload(), "Request interceptor returns an error");
+}
+
+final http:Client requestErrorInterceptorClientEP = check new("http://localhost:" + requestErrorInterceptorTestPort.toString());
+
+listener http:Listener requestErrorInterceptorServerEP = new(requestErrorInterceptorTestPort, config = {
+        interceptors : [new DefaultRequestInterceptor(), new RequestInterceptorReturnsError(), new DefaultRequestErrorInterceptor(), new LastRequestInterceptor()]
+    });
```

Comment:
Formatting issue.

Reply:
Fixed in [`1100634`](https://github.com/ballerina-platform/module-ballerina-http/pull/713/commits/110063402614b7d064c3a645e09751f3a3c772b5)